packages/lesswrong/server/codegen/generateFragmentTypes.ts

```diff
@@ -201,7 +205,9 @@ function getFragmentFieldType(fragmentName: string, parsedFragmentField: AnyBeca
     if (fieldName in schema) {
       const fieldSchema = schema[fieldName];
       assert(fieldSchema?.type);
-      if (fieldSchema?.resolveAs?.type && !fieldSchema?.resolveAs?.fieldName) {
+      if (fieldSchema?.typescriptType && fieldSchema?.blackbox) {
```

Comment:
I don't know if we want to keep this change anyways, but I don't think it's necessary to appropriately type the `glossary` post field.  That can be done by creating a simple-schema; see `topicPreferences` in `dialogueMatchPreferences > schema.ts`